app/src/main/java/com/duckduckgo/app/settings/SettingsViewModel.kt

```diff
@@ -82,17 +83,17 @@ class SettingsViewModel @Inject constructor(
         object LaunchLocation : Command()
         object LaunchWhitelist : Command()
         object LaunchAppIcon : Command()
-        object LaunchFireAnimationSettings : Command()
+        data class LaunchFireAnimationSettings(val animation: FireAnimation) : Command()
         object LaunchGlobalPrivacyControl : Command()
         object UpdateTheme : Command()
         object LaunchEmailDialog : Command()
+        data class ShowClearWhatDialog(val option: ClearWhatOption) : Command()
+        data class ShowClearWhenDialog(val option: ClearWhenOption) : Command()
     }
 
-    val viewState: MutableLiveData<ViewState> = MutableLiveData<ViewState>().apply {
-        value = ViewState()
-    }
+    private val viewState = MutableStateFlow(ViewState())
 
-    val command: SingleLiveEvent<Command> = SingleLiveEvent()
+    private val command = MutableSharedFlow<Command>()
```

Comment:
Not sure `SharedFlow` is the correct thing to use for commands. `SharedFlow` is hot so you could lose commands when there isn't an observer like during a configuration change.

Reply:
I think for command specifically we want the semantics to be those of share flow. Say a command is a toast or a dialog. We wouldn't want to show the toast every time we eg. rotate the screen.

Reply:
We are currently using these commands with `SingleLiveEvent` @marcosholgado. Otherwise we'd be using a normal `LiveData` for it. Personally, I'm ok with not showing the Toast again after a configuration change, specially because configuration changes in our app are very very rare.

Reply:
Yeah, it depends on the scenario really. For user commands it is ok but we also use commands for other things not directly related with a user action (and we probably shouldn't). I had a chat with @aitorvs earlier and we decided to go with a `Channel<Command>(capacity = 1, onBufferOverflow = BufferOverflow.DROP_OLDEST)` instead.